src/test/scala/lms/thirdparty/test_cuda.scala

```diff
@@ -164,5 +164,83 @@ class CudaTest extends TutorialFunSuite {
     System.out.println(indent(driver.code))
   }
 
+  test("kernel_reverse") {
+    val driver = new DslDriverCCuda[Int, Unit] {
+      
+      @virtualize
+      def snippet(arg: Rep[Int]) = {
+
+        val staticReverse = cudaGlobalFun[Array[Int], Int, Unit]((d, n) => {
+          val s = NewSharedArray[Int](64)
+          val t = threadIdxX
+          val tr = n - t - 1
+          s(t) = d(t)
+          cudaSyncThreads
+          d(t) = s(tr)
+        })
+
+        val dynamicReverse = cudaGlobalDynamicFun[Array[Int], Int, Unit]((d, n) => {
+          val s = NewDynSharedArray[Int]
+          val t = threadIdxX
+          val tr = n - t - 1
+          s(t) = d(t)
+          cudaSyncThreads
+          d(t) = s(tr)
+        })
```

Comment:
let's also move these 2 functions to cuda.scala (as a library that every other test can use if it wants to)